### PR TITLE
MQTTS done right

### PR DESCRIPTION
**Pull request by BadWolf42**
Added TLS support and Client Cert/Key support
Added special Paho logging functions in Python daemon
Using "TLS Secure" naming instead of "TLS Insecure" Paho naming
Automatic default MQTT port selection if TLS or not
Documentation and Ortho/Grammar
Safer queue in jeedom_socket class
Rework jMqttClient & MqttClient to avoid Exception in __init__
Removed whitespaces at the end of lines
TODO Add cert upload button
TODO Dynamically populate fTlsCaFile, fTlsClientCertFile & fTlsClientKeyFile dropdown lists (currently generated in php at page gen)
#TODO check_associate_cert_with_private_key
Bug fixes, doc and cosmetics.

### CODE REVIEW

File: core/class/jMQTTBase.class.php [hunk 1]
```diff
@@ -176,16 +176,37 @@ protected static function send_to_mqtt_daemon($params) {
...
       $params['id']=$id;
       $params['callback']='ws://127.0.0.1:'.config::byKey('websocketport', get_called_class(), get_called_class()::DEFAULT_WEBSOCKET_PORT).'/plugins/jMQTT/core/php/jmqttd.php';
       $params['hostname']=$hostname;
-      $params['port']=$port;
       $params['clientid']=$clientid;
       $params['statustopic']=$statustopic;
       $params['username']=$username;
       $params['password']=$password;
+      if ($tlsclicertfile != '')
+         $params['tlsclicertfile']=realpath(dirname(__FILE__) . '/../../data/certs/'.$tlsclicertfile);
```
Comment (Domochip) [inline core/class/jMQTTBase.class.php]: I would prefer that only absolute path is passed to this function. Path construction needs to be done in jMQTT.class.php

File: docs/fr_FR/index.md [hunk 2]
```diff
@@ -32,20 +32,20 @@ Après installation du plugin, l’activer. Celui-ci prend quelques minutes pour
...
-![Gestion plugin et brokers](../images/gestion_plugin_et_brokers.png)
+Le panneau supérieur gauche, intitulé *Gestion plugin et Broker*, permet de configurer le plugin, et de lister les équipements Broker.
+![Gestion plugin et Broker](../images/gestion_plugin_et_Broker.png)
 
-Sur les équipements broker, un point de couleur indique l'état de la connexion au broker :  
-![Status broker](../images/broker_status.png)  
-* Vert: la connexion au broker est opérationnelle
-* Orange: le démon tourne mais la connexion au broker n'est pas établie
+Sur les équipements Broker, un point de couleur indique l'état de la connexion au Broker :
+![Status Broker](../images/Broker_status.png)
```
Comment (Domochip) [inline docs/fr_FR/index.md]: filename case changed

File: resources/jmqttd/jeedom/jeedom.py [hunk 3]
```diff
@@ -157,7 +157,7 @@ def convert_log_level(level = 'error'):
 
 	@staticmethod
 	def set_log_level(level = 'error'):
-		FORMAT = '[%(asctime)-15s][%(levelname)s] : %(message)s'
+		FORMAT = '[%(asctime)-15s][%(levelname) 7s] : %(message)s'
```
Comment (Domochip) [inline resources/jmqttd/jeedom/jeedom.py]: lot of examples use this format for log level : [%(levelname)-8s]
Isn't it better than [%(levelname) 7s]
ref : https://docs.python.org/3/howto/logging-cookbook.html

File: core/class/jMQTTBase.class.php (hunk 1 above)
Comment (BadWolf42) [reply core/class/jMQTTBase.class.php]: Can I rework new_mqtt_client() function?
It has way too much parameters, can I pass $params directly in?
I'll suggest a prototype like:
```
public static function new_mqtt_client($id,               $hostname, 
                                       $port = 0,         $clientid = '',
                                       $statustopic = '', $params = array())
```
or even just:
```
public static function new_mqtt_client($id, $hostname, $params = array())
```

Comment (Domochip) [reply core/class/jMQTTBase.class.php]: yes

Comment (Domochip) [reply core/class/jMQTTBase.class.php]: second one is better :-)

File: resources/jmqttd/jeedom/jeedom.py (hunk 3 above)
Comment (BadWolf42) [reply resources/jmqttd/jeedom/jeedom.py]: All right done !

File: docs/fr_FR/index.md [hunk 4]
```diff
@@ -32,20 +32,20 @@ Après installation du plugin, l’activer. Celui-ci prend quelques minutes pour
...
 jMQTT posséde 2 types d'équipements:
-  - les **équipements de type broker** gérant la connexion avec les brokers MQTT, appelés équipements broker dans la suite;
-  - les **équipements "classiques"**, appélés simplement équipements dans la suite, 
+  - les **équipements de type Broker** gérant la connexion avec les Broker MQTT, appelés équipements Broker dans la suite;
+  - les **équipements "classiques"**, appélés simplement équipements dans la suite,
 
-Le panneau supérieur gauche, intitulé *Gestion plugin et brokers*, permet de configurer le plugin, et de lister les équipements broker.  
-![Gestion plugin et brokers](../images/gestion_plugin_et_brokers.png)
+Le panneau supérieur gauche, intitulé *Gestion plugin et Broker*, permet de configurer le plugin, et de lister les équipements Broker.
+![Gestion plugin et Broker](../images/gestion_plugin_et_Broker.png)
```
[the remark this answers is not in the document]
Comment (BadWolf42) [reply docs/fr_FR/index.md]: my bad, fixed

File: docs/fr_FR/index.md (hunk 2 above)
Comment (BadWolf42) [reply docs/fr_FR/index.md]: my bad, fixed

**Comment by Domochip** [comment]
It was long, challenging (because I'm changing too much things in between), but YES, here we are!!!
Thanks a lot for all your help 🤝

**Comment by AlexpFr** [comment]
congrats @BadWolf42 and @Domochip !

**Pull request merged**